Platform/CommonBoardPkg/Library/TccLib/TccLib.c

```diff
@@ -0,0 +1,639 @@
+
+#define SIGNATURE_IPFW    SIGNATURE_32 ('I', 'P', 'F', 'W')
+#define SIGNATURE_TCCT    SIGNATURE_32 ('T', 'C', 'C', 'T')
+#define SIGNATURE_TCCB    SIGNATURE_32 ('T', 'C', 'C', 'B')
+#define SIGNATURE_TCCC    SIGNATURE_32 ('T', 'C', 'C', 'C')
+#define SIGNATURE_TCCM    SIGNATURE_32 ('T', 'C', 'C', 'M')
+
+TCC_INFO_HOB    *mTccInfoHob    = NULL;
+BIOS_SETTINGS   *PolicyConfig;
+TCC_BINARIES_BASE *TccBinaries;
+UINT8 mTccRtd3Support;
+UINT8 mTccLowPowerS0Idle;
+
+/**
+  Update FSP-M UPD config data for TCC tuning, load TCC buffer & tuning data binaries from TempRam
+**/
+
+EFI_STATUS
+EFIAPI
+TccModePreMemConfig (
+  OUT BIOS_SETTINGS*     PreMemPolicyConfig,
+  OUT TCC_BINARIES_BASE* PreMemTccBinaries
+)
+{
+  TCC_STREAM_CONFIGURATION              *PreMemStreamConfig;
+  EFI_STATUS                            Status;
+  VOID                                  *TccPreMemBufferBase;
+  UINT32                                TccPreMemBufferSize;
+  VOID                                  *TccPreMemTuningBase;
+  UINT32                                TccPreMemTuningSize;
+
+  DEBUG ((DEBUG_INFO, "TccModePreMemConfig () - Start\n"));
+
+  // Load TCC tuning data from container
+  Status = LoadComponent (SIGNATURE_IPFW, SIGNATURE_TCCT, &TccPreMemTuningBase, &TccPreMemTuningSize);
+  if (EFI_ERROR (Status) || AsciiStrCmp ((CHAR8*) TccPreMemTuningBase, "dummy\0") == 0 ||
+      TccPreMemTuningSize < sizeof (TCC_STREAM_CONFIGURATION)) {
+    // Set default tuning values for TCC mode if TCC Tuning binary is not found
+    DEBUG ((DEBUG_INFO, "TCC Tuning data not found! Using default TCC tuning policies.\n"));
+    PreMemPolicyConfig               = NULL;
+    TccPreMemTuningBase        = NULL;
+    TccPreMemTuningSize        = 0;
+
+  } else {
+    // Consume TCC tuning binary settings
+    PreMemStreamConfig = (TCC_STREAM_CONFIGURATION *) TccPreMemTuningBase;
+    PreMemPolicyConfig = (BIOS_SETTINGS *) &PreMemStreamConfig->Info.Format1.BiosSettings;
+  }
+
+  // Load Tcc buffers data from container
+  Status = LoadComponent (SIGNATURE_IPFW, SIGNATURE_TCCB, &TccPreMemBufferBase, &TccPreMemBufferSize);
+  if (EFI_ERROR (Status) || AsciiStrCmp ((CHAR8*) TccPreMemBufferBase, "dummy\0") == 0) {
+    DEBUG ((DEBUG_INFO, "TCC  Buffer not found! %r\n", Status));
+    TccPreMemBufferBase = NULL;
+    TccPreMemBufferSize = 0;
+  }
+
+  // Assign the Tcc buffer & tuning to FSP M UPDs
+  PreMemTccBinaries->BufferBase       = TccPreMemBufferBase;
+  PreMemTccBinaries->BufferSize       = TccPreMemBufferSize;
+  PreMemTccBinaries->TuningBase       = TccPreMemTuningBase;
+  PreMemTccBinaries->TuningSize       = TccPreMemTuningSize;
+
+  DEBUG ((DEBUG_INFO, "TccModePreMemConfig () - End\n"));
+  return EFI_SUCCESS;
+}
+
+/**
+  Update FSP-S UPD config data for TCC tuning, load TCC related binaries into FSP
+**/
+
+EFI_STATUS
+EFIAPI
+TccModePostMemConfig (
+  VOID
+)
+{
+  TCC_STREAM_CONFIGURATION              *StreamConfig;
+  EFI_STATUS                            Status;

```

Comment:
I prefer to have this line at the end of all the declarations. Just a styling readability.